spectral_cube/spectral_cube.py

```diff
@@ -2363,6 +2363,8 @@ def __init__(self, data, wcs, mask=None, meta=None, fill_value=np.nan,
             self.pixels_per_beam = (self.beam.sr /
                                     (astropy.wcs.utils.proj_plane_pixel_area(self.wcs) *
                                      u.deg**2)).to(u.dimensionless_unscaled).value
+        self.abort = False  # Checked mid calculation
+        self.external_update_function = None  # External function that updates a gui's progress bar
```

Comment:
Rather than have this as a class attribute, why not just make it a keyword argument to the function?

Reply:
If we make these variables local to the function, we would not be able to externally change their states.